petri_dish/app.py

```diff
@@ -75,6 +77,17 @@ def get_all_subjects(self):
         """
         new_subjects = self.subject_source.read()
         grouped_subjects = self.subject_sink.read()
+        if grouped_subjects is None:
+            grouped_subjects = pandas.DataFrame(
+                columns=[
+                    self.index_column_name,
+                    # XXX: These columns need to be added to the DF even if
+                    # they're absent.
+                    self.GROUP_COLUMN_NAME,
+                    self.STAGE_COLUMN_NAME,
+                    self.JOINED_COLUMN_NAME,
```

Comment:
Can you remind me what is the purpose of this column?

Reply:
It's the date the subject joined the experiment. I'll make a note to document it.

Reply:
brilliant, thank you